cli/flags.rs

```diff
@@ -1487,6 +1511,40 @@ fn ca_file_arg_parse(flags: &mut Flags, matches: &clap::ArgMatches) {
   flags.ca_file = matches.value_of("cert").map(ToOwned::to_owned);
 }
 
+fn location_arg_parse(
+  flags: &mut Flags,
+  matches: &clap::ArgMatches,
+) -> clap::Result<()> {
+  flags.location = match matches.value_of("location") {
+    None => None,
+    Some(href) => Some(
+      Url::parse(href)
+        .map_err(|e| e.to_string())
+        .and_then(|url| {
+          if !["http", "https", "file"].contains(&url.scheme()) {
+            Err("Expected protocol \"http\", \"https\" or \"file\"".to_string())
+          } else {
+            Ok(url)
+          }
+        })
+        .map_err(|s| clap::Error {
+          kind: clap::ErrorKind::InvalidValue,
+          message: format!("Invalid location href \"{}\": {}", href, s),
+          info: None,
+        })?,
```

Comment:
Can this function be turned into `clap` validator function in `Arg` declaration?